### PR TITLE
support additionalItems as a schema for array validation

### PR DESCRIPTION
**Pull request by jkamerer**

### CODE REVIEW

File: lib/json_schema/validator.rb [hunk 1]
```diff
@@ -249,6 +249,12 @@ def validate_items(schema, data, errors, path)
           false
         else
           valid = true
+          if data.size > schema.items.count && schema.additional_items.is_a?(Schema)
+            (schema.items.count..data.count-1).each do |i|
```
Comment (brandur) [inline lib/json_schema/validator.rb]: Kind of a nit, but would you mind using `data.count-1` (i.e. spaces around the minus) just to conform with the existing convention here? Thanks!

File: lib/json_schema/validator.rb [hunk 2]
```diff
@@ -249,6 +249,12 @@ def validate_items(schema, data, errors, path)
...
+              valid = strict_and valid,
+                validate_data(schema.additional_items, data[i], errors, path + [i])
+            end
+					end
```
Comment (brandur) [inline lib/json_schema/validator.rb]: Probably an editor gone haywire, but your `end` on this line has migrated out to the frontier. Would you mind tightening that up?

**Comment by brandur** [comment]
@jkamerer Wow, amazing work here! Thanks a lot for adding the comprehensive test suite in particular. I've added a few minor comments above, but once they're fixed this is good to come in.

I'm not sure how I missed the schema version of `additionalItems` the first time around. Crazy!

File: lib/json_schema/validator.rb (hunk 1 above)
Comment (jkamerer) [inline lib/json_schema/validator.rb]: i appreciate the attention to detail and have no problem fixing that.

File: lib/json_schema/validator.rb (hunk 2 above)
Comment (jkamerer) [inline lib/json_schema/validator.rb]: Whoops! I was trying to stick to your space indentation convention, but i usually indent with tabs and have my editor set up to show two space tabs, so it all looked the same in my editor. I meant to convert all tabs to spaces before committing but obviously missed a spot.

**Comment by jkamerer** [comment]
@brandur thanks for checking out the pull request so quickly. I have this monkey patched locally but will be nice to update the gem and get rid of the monkey patch. This use of additionalItems in this way is possibly not that common, but I need it for a schema definition I'm using (obviously). At least your parser was allowing a Hash, so you noticed it at some level, just didn't follow through all the way implement it!

**Comment by brandur** [comment]
@jkamerer Thanks for the fixes!

Merged and released as 0.13.0.

**Pull request merged**